tests/test.c

```diff
@@ -138,10 +138,32 @@ void end_test() {
     printf("************************************** Running Checks... ******************************************\n");
 }
 
+//Implementation of strstr but with a regex string for needle
+char *rstrstr(const char *haystack, const char *needle) {
+    regex_t expr;
+    regmatch_t tracker;
+
+    if (regcomp(&expr, needle, REG_EXTENDED)) {
+        fprintf(stderr, "Unable to build regex expression");
+        return NULL;
+    }
+    if (regexec(&expr, haystack, 1, &tracker, REG_EXTENDED)) {
+        return NULL;
+    }
+    return haystack + tracker.rm_so;
+}
+
 // Verifies that expected_output is somewhere in the output
-void in_output(char* expected_output) {
+void in_output(char* expected_output, const uint8_t USE_REGEX) {
+    char *(*checker)(const char *, const char *); 
```

Comment:
Daniel and I were thinking that while using a function pointer here is perfectly valid, it may not be as readable for the general reader! I suggest using something like
```
// Do check
good_check = NULL
if (use_regex)
    good_check = rstrstr(...)
else
    good_check = strstr(...)
// Output result
if (good_check != NULL)
....
```

Reply:
Yea I agree. The suggested code might make things more readable.